### ISSUE TITLE
implement a cleanup function (_fini?) instead of manually invoking fflush(stdout)

### ISSUE THREAD

**Issue by lunixbochs** [lunixbochs/lib43#14]
Going forward, I want to think about the ability of a lib43 user to tune the runtime. One of the most useful cases right now is beating the startup/teardown cost of _every other_ libc and crt0 I've tested, sometimes by a significant margin.

Adding a `_fini` function that does anything like fflush() on all file descriptors could waste even precious _nanoseconds_ which is useful if using lib43 for competitive reasons.

**Comment by lunixbochs** [comment]
Probably either `cmake -DUNSAFE_BUT_FAST=1` or individual flags like `cmake -DFLUSH_WRITES=1`

**Comment by ghost** [comment]
Agreed but it should maybe be tunable on a by feature basis in an organized way. Same arguments could apply to argv passing in some cases and other things as well.

**Comment by lunixbochs** [comment]
setting up argv is like two instructions, while "flush all open file handles" is actually a ton of work and potentially many syscalls.

**Comment by ghost** [comment]
I realize that - my point is just that there may be other things people may want to turn on and off as more stuff goes into library so having a methodical way to do so is good. Unwrapping argv is potentially more depending on what compiler decides to do for function prologue but it still wouldn't be that expensive true.
